README.md
***

<div align="center">

[![Build Status](https://travis-ci.org/magx2/jSuplaServerMock.svg?branch=master)](https://travis-ci.org/magx2/jSuplaServerMock)
[![MIT license](http://img.shields.io/badge/license-MIT-brightgreen.svg?style=flat)](http://opensource.org/licenses/MIT)
[![Download](https://api.bintray.com/packages/big-boy/bigboy/jSuplaServerMock/images/download.svg) ](https://bintray.com/big-boy/bigboy/jSuplaServerMock/_latestVersion)

This server was generated by the [swagger-codegen](https://github.com/swagger-api/swagger-codegen) project.

</div>

***

# Swagger Generated Server

jSupla Mock Server allows you to start server that is compatible with Supla Cloud. By default it has mock all possible devices.  

# Run Server

Download latest version from [![Download](https://api.bintray.com/packages/big-boy/bigboy/jSuplaServerMock/images/download.svg) ](https://bintray.com/big-boy/bigboy/jSuplaServerMock/_latestVersion) and then run

```$bash
java -jar jsupla-mock-server-<VERSION>.jar

```

After about 30 seconds server should be started. To test it you can send test request to `http://localhost:9090/api/2.3.0/server-info`

```bash
curl -X GET "http://localhost:9090/api/v2.3.0/server-info" -H  "accept: application/json"
```

As a response you should get something like this (not formatted):

```json
{
  "address": "http://localhost:9090/api/v2.3.0",
  "time": null,
  "timezone": null,
  "authenticated": null,
  "username": null,
  "cloudVersion": "2.3.0",
  "apiVersion": "2.3.0",
  "supportedApiVersions": [
	"2.3.0"
  ]
} 
```

For more api calls you can check [Supla Swagger Hub](https://app.swaggerhub.com/apis/supla/supla-cloud-api/2.3.0#/Server/getServerInfo)

## Authorization

Most of calls need to be authorized with oAuth token. The default token is `MzFhYTNiZTAwODg5M2E0NDE3OGUwNWE5ZjYzZWQ2YzllZGFiYWRmNDQwNDBlNmZhZGEzN2I3NTJiOWM2ZWEyZg`. You can change it in `/src/main/resources/application.yml` or during startup via Spring Boot properties (`--jsupla.token=your_token`).

_Note 1: Tokens generated from Supla Cloud are made from 2 parts that are separated with dot. First part is plain OAuth token. Second part is server URL encoded in Base64. In this implementation you don't need to pass anything as server URL, but if you will pass it, it will be trimmed._    

_Note 2: If you want to use token that will point you to `http://localhost:9090` you can use this token `MzFhYTNiZTAwODg5M2E0NDE3OGUwNWE5ZjYzZWQ2YzllZGFiYWRmNDQwNDBlNmZhZGEzN2I3NTJiOWM2ZWEyZg.aHR0cDovL2xvY2FsaG9zdDo5MDkw`_

# Gradle & Maven

To embed Supla server in you Gradle/Maven application you first need to add `big-boy` bintray repository: 

```groovy
repositories { 
	maven { 
		url "https://dl.bintray.com/big-boy/bigboy" 
	} 
}
```

```xml
<repositories>
	<repository>
		<snapshots>
			<enabled>false</enabled>
		</snapshots>
		<id>bintray-big-boy-bigboy</id>
		<name>bintray</name>
		<url>https://dl.bintray.com/big-boy/bigboy</url>
	</repository>
</repositories>
```

After this just add dependency:

```groovy
compile 'pl.grzeslowski.jsuplaservermock:jsupla-mock-server:VERSION'
```

```xml
<dependency>
  <groupId>pl.grzeslowski.jsuplaservermock</groupId>
  <artifactId>jsupla-mock-server</artifactId>
  <version>VERSION</version>
  <type>pom</type>
</dependency>
```

The latest version you can find here [![Download](https://api.bintray.com/packages/big-boy/bigboy/jSuplaServerMock/images/download.svg) ](https://bintray.com/big-boy/bigboy/jSuplaServerMock/_latestVersion)


docs/release-notes.md
<sup><sup>*Release notes were automatically generated by [Shipkit](http://shipkit.org/)*</sup></sup>

#### 0.9.26
 - 2019-09-10 - [3 commits](https://github.com/magx2/jSuplaServerMock/compare/v0.9.25...v0.9.26) by [Martin Grzeslowski](https://github.com/magx2) - published to [![Bintray](https://img.shields.io/badge/Bintray-0.9.26-green.svg)](https://bintray.com/big-boy/bigboy/jSuplaServerMock/0.9.26)
 - No pull requests referenced in commit messages.

#### 0.9.25
 - 2019-08-14 - [1 commit](https://github.com/magx2/jSuplaServerMock/compare/v0.9.24...v0.9.25) by [Martin Grzeslowski](https://github.com/magx2) - published to [![Bintray](https://img.shields.io/badge/Bintray-0.9.25-green.svg)](https://bintray.com/big-boy/bigboy/jSuplaServerMock/0.9.25)
 - No pull requests referenced in commit messages.

#### 0.9.24
 - 2019-08-14 - [4 commits](https://github.com/magx2/jSuplaServerMock/compare/v0.9.23...v0.9.24) by [Martin Grzeslowski](https://github.com/magx2) - published to [![Bintray](https://img.shields.io/badge/Bintray-0.9.24-green.svg)](https://bintray.com/big-boy/bigboy/jSuplaServerMock/0.9.24)
 - No pull requests referenced in commit messages.

#### 0.9.23
 - 2019-08-13 - [3 commits](https://github.com/magx2/jSuplaServerMock/compare/v0.9.22...v0.9.23) by [Martin Grzeslowski](https://github.com/magx2) - published to [![Bintray](https://img.shields.io/badge/Bintray-0.9.23-green.svg)](https://bintray.com/big-boy/bigboy/jSuplaServerMock/0.9.23)
 - No pull requests referenced in commit messages.

#### 0.9.22
 - 2019-08-09 - [1 commit](https://github.com/magx2/jSuplaServerMock/compare/v0.9.21...v0.9.22) by [Martin Grzeslowski](https://github.com/magx2) - published to [![Bintray](https://img.shields.io/badge/Bintray-0.9.22-green.svg)](https://bintray.com/big-boy/bigboy/jSuplaServerMock/0.9.22)
 - No pull requests referenced in commit messages.

#### 0.9.21
 - 2019-08-09 - [2 commits](https://github.com/magx2/jSuplaServerMock/compare/v0.9.20...v0.9.21) by [Martin Grzeslowski](https://github.com/magx2) - published to [![Bintray](https://img.shields.io/badge/Bintray-0.9.21-green.svg)](https://bintray.com/big-boy/bigboy/jSuplaServerMock/0.9.21)
 - No pull requests referenced in commit messages.

#### 0.9.20
 - 2019-08-04 - [1 commit](https://github.com/magx2/jSuplaServerMock/compare/v0.9.19...v0.9.20) by [Martin Grzeslowski](https://github.com/magx2) - published to [![Bintray](https://img.shields.io/badge/Bintray-0.9.20-green.svg)](https://bintray.com/big-boy/bigboy/jSuplaServerMock/0.9.20)
 - Add random changing [(#38)](https://github.com/magx2/jSuplaServerMock/issues/38)

#### 0.9.19
 - 2019-08-04 - [1 commit](https://github.com/magx2/jSuplaServerMock/compare/v0.9.18...v0.9.19) by [Martin Grzeslowski](https://github.com/magx2) - published to [![Bintray](https://img.shields.io/badge/Bintray-0.9.19-green.svg)](https://bintray.com/big-boy/bigboy/jSuplaServerMock/0.9.19)
 - Add random changing [(#38)](https://github.com/magx2/jSuplaServerMock/issues/38)

#### 0.9.18
 - 2019-08-04 - [1 commit](https://github.com/magx2/jSuplaServerMock/compare/v0.9.17...v0.9.18) by [Martin Grzeslowski](https://github.com/magx2) - published to [![Bintray](https://img.shields.io/badge/Bintray-0.9.18-green.svg)](https://bintray.com/big-boy/bigboy/jSuplaServerMock/0.9.18)
 - Add random changing [(#38)](https://github.com/magx2/jSuplaServerMock/issues/38)

#### 0.9.17
 - 2019-08-04 - [1 commit](https://github.com/magx2/jSuplaServerMock/compare/v0.9.16...v0.9.17) by [Martin Grzeslowski](https://github.com/magx2) - published to [![Bintray](https://img.shields.io/badge/Bintray-0.9.17-green.svg)](https://bintray.com/big-boy/bigboy/jSuplaServerMock/0.9.17)
 - Add random changing [(#38)](https://github.com/magx2/jSuplaServerMock/issues/38)

#### 0.9.16
 - 2019-08-04 - [6 commits](https://github.com/magx2/jSuplaServerMock/compare/v0.9.15...v0.9.16) by [Martin Grzeslowski](https://github.com/magx2) - published to [![Bintray](https://img.shields.io/badge/Bintray-0.9.16-green.svg)](https://bintray.com/big-boy/bigboy/jSuplaServerMock/0.9.16)
 - Add random changing [(#38)](https://github.com/magx2/jSuplaServerMock/issues/38)

#### 0.9.15
 - 2019-08-03 - [2 commits](https://github.com/magx2/jSuplaServerMock/compare/v0.9.14...v0.9.15) by [Martin Grzeslowski](https://github.com/magx2) - published to [![Bintray](https://img.shields.io/badge/Bintray-0.9.15-green.svg)](https://bintray.com/big-boy/bigboy/jSuplaServerMock/0.9.15)
 - [Enhancement] Implement endpoint `devices/id/channels` [(#36)](https://github.com/magx2/jSuplaServerMock/issues/36)

#### 0.9.14
 - 2019-06-16 - [3 commits](https://github.com/magx2/jSuplaServerMock/compare/v0.9.13...v0.9.14) by [Martin Grzeslowski](https://github.com/magx2) - published to [![Bintray](https://img.shields.io/badge/Bintray-0.9.14-green.svg)](https://bintray.com/big-boy/bigboy/jSuplaServerMock/0.9.14)
 - [Bug] Add `callibraiting` to roller shutter channel [(#35)](https://github.com/magx2/jSuplaServerMock/issues/35)
 - [Bug] Add `hi` to roller shutter channel [(#34)](https://github.com/magx2/jSuplaServerMock/issues/34)
 - [Bug] Add `connected` property to channel [(#33)](https://github.com/magx2/jSuplaServerMock/issues/33)

#### 0.9.13
 - 2019-06-15 - [2 commits](https://github.com/magx2/jSuplaServerMock/compare/v0.9.12...v0.9.13) by [Martin](https://github.com/magx2) - published to [![Bintray](https://img.shields.io/badge/Bintray-0.9.13-green.svg)](https://bintray.com/big-boy/bigboy/jSuplaServerMock/0.9.13)
 - [Enhancement] #8 Implement location endpoint [(#32)](https://github.com/magx2/jSuplaServerMock/pull/32)
 - [Enhancement] Implement `Locations` endpoints [(#8)](https://github.com/magx2/jSuplaServerMock/issues/8)

#### 0.9.12

src/main/java/io/swagger/model/Device.java
package io.swagger.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.swagger.annotations.ApiModelProperty;
import org.springframework.validation.annotation.Validated;
import org.threeten.bp.OffsetDateTime;

import javax.validation.Valid;
import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

/**
 * Device
 */
@Validated
@javax.annotation.Generated(value = "io.swagger.codegen.v3.generators.java.SpringCodegen", date = "2019-04-15T15:14:44.820Z[GMT]")
public class Device {
    @JsonProperty("id")
    private Integer id = null;

    @JsonProperty("name")
    private String name = null;

    @JsonProperty("enabled")
    private Boolean enabled = null;

    @JsonProperty("comment")
    private String comment = null;

    @JsonProperty("regDate")
    private OffsetDateTime regDate = null;

    @JsonProperty("regIpv4")
    private Long regIpv4 = null;

    @JsonProperty("lastConnected")
    private OffsetDateTime lastConnected = null;

    @JsonProperty("lastIpv4")
    private Long lastIpv4 = null;

    @JsonProperty("softwareVersion")
    private String softwareVersion = null;

    @JsonProperty("gUIDString")
    private String gUIDString = null;

    @JsonProperty("locationId")
    private Integer locationId = null;

    @JsonProperty("originalLocationId")
    private Integer originalLocationId = null;

    @JsonProperty("channelsIds")
    @Valid
    private List<Integer> channelsIds = null;

    @JsonProperty("connected")
    private Boolean connected = null;

    @JsonProperty("location")
    private Location location = null;

    @JsonProperty("originalLocation")
    private Location originalLocation = null;

    @JsonProperty("channels")
    @Valid
    private List<Channel> channels = null;

    @JsonProperty("schedules")
    @Valid
    private List<Schedule> schedules = null;

    public Device id(Integer id) {
        this.id = id;
        return this;
    }

    /**
     * Device identifier
     *
     * @return id
     **/
    @ApiModelProperty(value = "Device identifier")

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public Device name(String name) {
        this.name = name;
        return this;
    }

    /**
     * Get name
     *
     * @return name
     **/
    @ApiModelProperty(value = "")

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Device enabled(Boolean enabled) {
        this.enabled = enabled;
        return this;
    }

    /**
     * Get enabled
     *
     * @return enabled
     **/
    @ApiModelProperty(value = "")

    public Boolean isEnabled() {
        return enabled;
    }

    public void setEnabled(Boolean enabled) {
        this.enabled = enabled;
    }

    public Device comment(String comment) {
        this.comment = comment;
        return this;
    }

    /**
     * Get comment
     *
     * @return comment
     **/
    @ApiModelProperty(value = "")

    public String getComment() {
        return comment;
    }

    public void setComment(String comment) {
        this.comment = comment;
    }

    public Device regDate(OffsetDateTime regDate) {
        this.regDate = regDate;
        return this;
    }

    /**
     * Get regDate
     *
     * @return regDate
     **/
    @ApiModelProperty(value = "")

    @Valid
    public OffsetDateTime getRegDate() {
        return regDate;
    }

    public void setRegDate(OffsetDateTime regDate) {
        this.regDate = regDate;
    }

    public Device regIpv4(Long regIpv4) {
        this.regIpv4 = regIpv4;
        return this;
    }

    /**
     * Get regIpv4
     *
     * @return regIpv4
     **/
    @ApiModelProperty(value = "")

    public Long getRegIpv4() {
        return regIpv4;
    }

    public void setRegIpv4(Long regIpv4) {
        this.regIpv4 = regIpv4;
    }

    public Device lastConnected(OffsetDateTime lastConnected) {
        this.lastConnected = lastConnected;
        return this;
    }

    /**
     * Get lastConnected
     *
     * @return lastConnected
     **/
    @ApiModelProperty(value = "")

    @Valid
    public OffsetDateTime getLastConnected() {
        return lastConnected;
    }

    public void setLastConnected(OffsetDateTime lastConnected) {
        this.lastConnected = lastConnected;
    }

    public Device lastIpv4(Long lastIpv4) {
        this.lastIpv4 = lastIpv4;
        return this;
    }

    /**
     * Get lastIpv4
     *
     * @return lastIpv4
     **/
    @ApiModelProperty(value = "")

    public Long getLastIpv4() {
        return lastIpv4;
    }

    public void setLastIpv4(Long lastIpv4) {
        this.lastIpv4 = lastIpv4;
    }

    public Device softwareVersion(String softwareVersion) {
        this.softwareVersion = softwareVersion;
        return this;
    }

    /**
     * Get softwareVersion
     *
     * @return softwareVersion
     **/
    @ApiModelProperty(value = "")

    public String getSoftwareVersion() {
        return softwareVersion;
    }

    public void setSoftwareVersion(String softwareVersion) {
        this.softwareVersion = softwareVersion;
    }

    public Device gUIDString(String gUIDString) {
        this.gUIDString = gUIDString;
        return this;
    }

    /**
     * Get gUIDString
     *
     * @return gUIDString
     **/
    @ApiModelProperty(value = "")

    public String getGUIDString() {
        return gUIDString;
    }

    public void setGUIDString(String gUIDString) {
        this.gUIDString = gUIDString;
    }

    public Device locationId(Integer locationId) {
        this.locationId = locationId;
        return this;
    }

    /**
     * Get locationId
     *
     * @return locationId
     **/
    @ApiModelProperty(value = "")

    public Integer getLocationId() {
        return locationId;
    }

    public void setLocationId(Integer locationId) {
        this.locationId = locationId;
    }

    public Device originalLocationId(Integer originalLocationId) {
        this.originalLocationId = originalLocationId;
        return this;
    }

    /**
     * Get originalLocationId
     *
     * @return originalLocationId
     **/
    @ApiModelProperty(value = "")

    public Integer getOriginalLocationId() {
        return originalLocationId;
    }

    public void setOriginalLocationId(Integer originalLocationId) {
        this.originalLocationId = originalLocationId;
    }

    public Device channelsIds(List<Integer> channelsIds) {
        this.channelsIds = channelsIds;
        return this;
    }

    public Device addChannelsIdsItem(Integer channelsIdsItem) {
        if (this.channelsIds == null) {
            this.channelsIds = new ArrayList<Integer>();
        }
        this.channelsIds.add(channelsIdsItem);
        return this;
    }

    /**
     * Get channelsIds
     *
     * @return channelsIds
     **/
    @ApiModelProperty(value = "")

    public List<Integer> getChannelsIds() {
        return channelsIds;
    }

    public void setChannelsIds(List<Integer> channelsIds) {
        this.channelsIds = channelsIds;
    }

    public Device connected(Boolean connected) {
        this.connected = connected;
        return this;
    }

    /**
     * Get connected
     *
     * @return connected
     **/
    @ApiModelProperty(value = "")

    public Boolean isConnected() {
        return connected;
    }

    public void setConnected(Boolean connected) {
        this.connected = connected;
    }

    public Device location(Location location) {
        this.location = location;
        return this;
    }

    /**
     * Get location
     *
     * @return location
     **/
    @ApiModelProperty(value = "")

    @Valid
    public Location getLocation() {
        return location;
    }

    public void setLocation(Location location) {
        this.location = location;
    }

    public Device originalLocation(Location originalLocation) {
        this.originalLocation = originalLocation;
        return this;
    }

    /**
     * Get originalLocation
     *
     * @return originalLocation
     **/
    @ApiModelProperty(value = "")

    @Valid
    public Location getOriginalLocation() {
        return originalLocation;
    }

    public void setOriginalLocation(Location originalLocation) {
        this.originalLocation = originalLocation;
    }

    public Device channels(List<Channel> channels) {
        this.channels = channels;
        return this;
    }

    public Device addChannelsItem(Channel channelsItem) {
        if (this.channels == null) {
            this.channels = new ArrayList<Channel>();
        }
        this.channels.add(channelsItem);
        return this;
    }

    /**
     * Returned only if requested by the `include` parameter.
     *
     * @return channels
     **/
    @ApiModelProperty(value = "Returned only if requested by the `include` parameter.")
    @Valid
    public List<Channel> getChannels() {
        return channels;
    }

    public void setChannels(List<Channel> channels) {
        this.channels = channels;
    }

    public Device schedules(List<Schedule> schedules) {
        this.schedules = schedules;
        return this;
    }

    public Device addSchedulesItem(Schedule schedulesItem) {
        if (this.schedules == null) {
            this.schedules = new ArrayList<Schedule>();
        }
        this.schedules.add(schedulesItem);
        return this;
    }

    /**
     * Returned only if requested by the `include` parameter.
     *
     * @return schedules
     **/
    @ApiModelProperty(value = "Returned only if requested by the `include` parameter.")
    @Valid
    public List<Schedule> getSchedules() {
        return schedules;
    }

    public void setSchedules(List<Schedule> schedules) {
        this.schedules = schedules;
    }

    @Override
    public boolean equals(java.lang.Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        Device device = (Device) o;
        return Objects.equals(this.id, device.id) &&
                       Objects.equals(this.name, device.name) &&
                       Objects.equals(this.enabled, device.enabled) &&
                       Objects.equals(this.comment, device.comment) &&
                       Objects.equals(this.regDate, device.regDate) &&
                       Objects.equals(this.regIpv4, device.regIpv4) &&
                       Objects.equals(this.lastConnected, device.lastConnected) &&
                       Objects.equals(this.lastIpv4, device.lastIpv4) &&
                       Objects.equals(this.softwareVersion, device.softwareVersion) &&
                       Objects.equals(this.gUIDString, device.gUIDString) &&
                       Objects.equals(this.locationId, device.locationId) &&
                       Objects.equals(this.originalLocationId, device.originalLocationId) &&
                       Objects.equals(this.channelsIds, device.channelsIds) &&
                       Objects.equals(this.connected, device.connected) &&
                       Objects.equals(this.location, device.location) &&
                       Objects.equals(this.originalLocation, device.originalLocation) &&
                       Objects.equals(this.channels, device.channels) &&
                       Objects.equals(this.schedules, device.schedules);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, name, enabled, comment, regDate, regIpv4, lastConnected, lastIpv4, softwareVersion, gUIDString, locationId, originalLocationId, channelsIds, connected, originalLocation, channels, schedules);
    }

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        sb.append("class Device {\n");

        sb.append("    id: ").append(toIndentedString(id)).append("\n");
        sb.append("    name: ").append(toIndentedString(name)).append("\n");
        sb.append("    enabled: ").append(toIndentedString(enabled)).append("\n");
        sb.append("    comment: ").append(toIndentedString(comment)).append("\n");
        sb.append("    regDate: ").append(toIndentedString(regDate)).append("\n");
        sb.append("    regIpv4: ").append(toIndentedString(regIpv4)).append("\n");
        sb.append("    lastConnected: ").append(toIndentedString(lastConnected)).append("\n");
        sb.append("    lastIpv4: ").append(toIndentedString(lastIpv4)).append("\n");
        sb.append("    softwareVersion: ").append(toIndentedString(softwareVersion)).append("\n");
        sb.append("    gUIDString: ").append(toIndentedString(gUIDString)).append("\n");
        sb.append("    locationId: ").append(toIndentedString(locationId)).append("\n");
        sb.append("    originalLocationId: ").append(toIndentedString(originalLocationId)).append("\n");
        sb.append("    channelsIds: ").append(toIndentedString(channelsIds)).append("\n");
        sb.append("    connected: ").append(toIndentedString(connected)).append("\n");
        sb.append("    originalLocation: ").append(toIndentedString(originalLocation)).append("\n");
        sb.append("    channels: ").append(toIndentedString(channels)).append("\n");
        sb.append("    schedules: ").append(toIndentedString(schedules)).append("\n");
        sb.append("}");
        return sb.toString();
    }

    /**
     * Convert the given object to string with each line indented by 4 spaces
     * (except the first line).
     */
    private String toIndentedString(java.lang.Object o) {
        if (o == null) {
            return "null";
        }
        return o.toString().replace("\n", "\n    ");
    }
}
